dataprofiler/profilers/profile_builder.py

```diff
@@ -1392,24 +1392,39 @@ def _get_correlation(self, clean_samples):
         corr_mat[rows, clean_column_ids] = np.corrcoef(data, rowvar=False)
         return corr_mat
 
-    def _update_correlation(self, clean_samples):
+    def _update_correlation(self, clean_samples, prev_dependent_properties):
         """
         Update correlation matrix for cleaned data.
 
         :param clean_samples: the input cleaned dataset
         :type clean_samples: dict()
         """
+        batch_corr = self._get_correlation(clean_samples)
         if self.total_samples > 0:
-            # currently return None with the existing correlation
-            # TODO: implement the update with the existing correlation
-            warnings.warn("Currently, the correlation update is disabled "
-                          "with existing correlation, which will be reset "
-                          "to None. Updating correlations will be available "
-                          "in a future update.")
-            self.correlation_matrix = None
+            data = pd.DataFrame(clean_samples).apply(pd.to_numeric, errors='coerce')
+
+            batch_means = np.full(len(self._profile.keys()), np.nan)
+            batch_stds = np.full(len(self._profile.keys()), np.nan)
+            for id, col in enumerate(self._profile.keys()):
+                if self._profile[col].profile['data_type'] in ['int', 'float'] \
+                        and col in data.columns:
+                    batch_means[id] = data[col].mean()
+                    batch_stds[id] = data[col].std()
+
+            # If these lengths are different, then some columns were added to to the
+            # profile.
+            if (len(batch_means) != len(prev_dependent_properties["mean"])):
+                warnings.warn("A new column was added when updating. Correlations cannot be calculated"
+                              " for new columns.", RuntimeWarning)
+                return
```

Comment:
Add a note to remove when @ChrisWallace2020 pr gets merged as it won't allow it.